File: backend/src/main/java/com/shopsphere/api/entity/Order.java
package com.shopsphere.api.entity;

import com.shopsphere.api.enums.OrderStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private LocalDateTime placedOn;

    private Double amount;

    @Enumerated(EnumType.STRING)
    private OrderStatus status;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "order_id")
    private List<OrderItem> items;

    private String estimatedDelivery;

    @Embedded
    private LogisticsInfo logistics;

    @Embedded
    @AttributeOverrides({
            @AttributeOverride(name = "street", column = @Column(name = "delivery_street")),
            @AttributeOverride(name = "city", column = @Column(name = "delivery_city")),
            @AttributeOverride(name = "state", column = @Column(name = "delivery_state")),
            @AttributeOverride(name = "zipCode", column = @Column(name = "delivery_zip_code")),
            @AttributeOverride(name = "country", column = @Column(name = "delivery_country"))
    })
    private Address deliveryAddress;
}

File: Frontend/src/app/models/order.ts

export type OrderStatus = 'Placed' | 'Confirmed' | 'Packed' | 'Shipped' | 'Delivered' | 'Cancelled';

export interface OrderItem {
  productId: string;
  name: string;
  image: string;
  quantity: number;
  color?: string;
  size?: string;
  material?: string;
  price: number;
}

export interface LogisticsInfo {
  carrier: string;
  trackingId: string;
  currentLocation: string;
}

export interface Order {
  id?: number;
  userId: number;
  placedOn: string;
  amount: number;
  status: OrderStatus;
  items: OrderItem[];
  estimatedDelivery: string;
  logistics: LogisticsInfo;
  deliveryAddress?: Address;
}

export interface Address {
  street: string;
  city: string;
  state: string;
  zipCode: string;
  country: string;
}
